### ISSUE TITLE
GenericName in .desktop file

### ISSUE THREAD

**Issue by Smarre** [fish-shell/fish-shell#7804]
This is more of a topic for discussion than anything else, but the fish.desktop file does not have GenericName declared, which at least OpenSUSE wants desktop files to have.

It’s a bit tricky to have a proper generic name, but I think “Command-line interpreter” would be pretty good one, which I found from [Unix shell wiki page](https://en.wikipedia.org/wiki/Unix_shell).

Is there any specific reason GenericName is not set?

**Comment by faho** [comment]
It's not set because the documentation on all this is fairly awkward so I didn't see it, and we really only need the .desktop file because the appimage requires one - having it actually show up in menus isn't really desirable because you'd start your terminal, not fish.

**Comment by Smarre** [comment]
Right, that makes sense. It’s not a problem, I just wanted to raise this topic since distributions often have their own conventions, which may or may not make sense for upstream project to adopt. 

I personally like the idea of desktop files, as it would allow searching applications without knowing the application’s name, which is useful for someone not familiar with how Linux works. For CLI applications, the benefit is of course smaller. At least on KDE, fish.desktop file can be used to open fish in Konsole from start menu.

**Comment by zanchey** [comment]
The Debian packages shipped with a menu entry for fish for as long as I can remember, and it does seem useful for those new to terminals/command lines.

**Comment by Exalm** [comment]
>  having it actually show up in menus isn't really desirable because you'd start your terminal, not fish.

You can add `NoDisplay=true` to the desktop file, then it won't show up in menus.
